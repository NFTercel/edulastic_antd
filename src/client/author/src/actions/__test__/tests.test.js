import {
  receiveTestsAction,
  receiveTestByIdAction,
  createTestAction,
  updateTestAction,
  setTestDataAction,
  setDefaultTestDataAction
} from "../tests";

describe("tests", () => {
  it("receive tests should return an action", () => {
    expect(receiveTestsAction(1, 5)).toMatchSnapshot();
    expect(receiveTestsAction(2, 5)).toMatchSnapshot();
    expect(receiveTestsAction(2, 10)).toMatchSnapshot();
  });

  it("receive test by id should return an action", () => {
    expect(receiveTestByIdAction("5bf2a3e0da9400105cac6cf6")).toMatchSnapshot();
    expect(receiveTestByIdAction("5bf2a3e0da9400105cac6cf7")).toMatchSnapshot();
    expect(receiveTestByIdAction("5bf2a3e0da9400105cac6cf8")).toMatchSnapshot();
  });

  it("create test should return an action", () => {
    expect(
      createTestAction({
        title: "New Test",
        description: "",
        renderingType: "assessment",
        status: "draft",
        thumbnail: "https://fakeimg.pl/500x135/",
        createdBy: {},
        tags: [],
        scoring: {
          total: 0,
          testItems: [
            {
              points: 0
            },
            {
              points: 0
            }
          ]
        },
        testItems: ["5bf28847dc39e409216aea00", "5bf28d4edc39e409216aea04"],
        assignments: [],
        standardsTag: {
          curriculum: "",
          standards: []
        },
        grades: [],
        subjects: [],
        courses: [],
        collections: "",
        analytics: {
          usage: "0",
          likes: "0"
        }
      })
    ).toMatchSnapshot();

    expect(
      createTestAction({
        title: "New Test2",
        description: "Unit test items",
        renderingType: "assessment",
        status: "draft",
        thumbnail: "https://fakeimg.pl/500x135/",
        createdBy: {},
        tags: [],
        scoring: {
          total: 0,
          testItems: [
            {
              points: 0
            },
            {
              points: 0
            }
          ]
        },
        testItems: ["5bf28847dc39e409216aea02", "5bf28d4edc39e409216aea06"],
        assignments: [],
        standardsTag: {
          curriculum: "",
          standards: []
        },
        grades: [],
        subjects: ["Math"],
        courses: ["1"],
        collections: "",
        analytics: {
          usage: "10",
          likes: "20"
        }
      })
    ).toMatchSnapshot();
  });

  it("update test should return an action", () => {
    expect(
      updateTestAction(
        ("5bf2a3e0da9400105cac6cf6",
        {
          title: "New Test",
          description: "",
          renderingType: "assessment",
          status: "draft",
          thumbnail: "https://fakeimg.pl/500x135/",
          createdBy: {},
          tags: [],
          scoring: {
            total: 0,
            testItems: [
              {
                points: 0
              },
              {
                points: 0
              }
            ]
          },
          testItems: ["5bf28847dc39e409216aea00", "5bf28d4edc39e409216aea04"],
          assignments: [],
          standardsTag: {
            curriculum: "",
            standards: []
          },
          grades: [],
          subjects: [],
          courses: [],
          collections: "",
          analytics: {
            usage: "0",
            likes: "0"
          }
        })
      )
    ).toMatchSnapshot();

    expect(
      updateTestAction(
        ("5bf2a3e0da9400105cac6cf6",
        {
          title: "New Test2",
          description: "Unit test items",
          renderingType: "assessment",
          status: "draft",
          thumbnail: "https://fakeimg.pl/500x135/",
          createdBy: {},
          tags: [],
          scoring: {
            total: 0,
            testItems: [
              {
                points: 0
              },
              {
                points: 0
              }
            ]
          },
          testItems: ["5bf28847dc39e409216aea02", "5bf28d4edc39e409216aea06"],
          assignments: [],
          standardsTag: {
            curriculum: "",
            standards: []
          },
          grades: [],
          subjects: ["Math"],
          courses: ["1"],
          collections: "",
          analytics: {
            usage: "10",
            likes: "20"
          }
        })
      )
    ).toMatchSnapshot();
  });

  it("set test data should return an action", () => {
    expect(
      setTestDataAction({
        sharing: [
          {
            email: "andrey@snapwiz.com",
            permission: "rxw"
          }
        ],
        createBy: {
          firstName: "andrey",
          lastName: "vasilev",
          email: "andrey@snapwiz.com"
        }
      })
    ).toMatchSnapshot();
  });

  it("set default test data should return an action", () => {
    expect(setDefaultTestDataAction()).toMatchSnapshot();
  });
});
