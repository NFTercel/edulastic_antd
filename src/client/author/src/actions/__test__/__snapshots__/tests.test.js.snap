// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests create test should return an action 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "analytics": Object {
        "likes": "0",
        "usage": "0",
      },
      "assignments": Array [],
      "collections": "",
      "courses": Array [],
      "createdBy": Object {},
      "description": "",
      "grades": Array [],
      "renderingType": "assessment",
      "scoring": Object {
        "testItems": Array [
          Object {
            "points": 0,
          },
          Object {
            "points": 0,
          },
        ],
        "total": 0,
      },
      "standardsTag": Object {
        "curriculum": "",
        "standards": Array [],
      },
      "status": "draft",
      "subjects": Array [],
      "tags": Array [],
      "testItems": Array [
        "5bf28847dc39e409216aea00",
        "5bf28d4edc39e409216aea04",
      ],
      "thumbnail": "https://fakeimg.pl/500x135/",
      "title": "New Test",
    },
  },
  "type": "[tests] create test request",
}
`;

exports[`tests create test should return an action 2`] = `
Object {
  "payload": Object {
    "data": Object {
      "analytics": Object {
        "likes": "20",
        "usage": "10",
      },
      "assignments": Array [],
      "collections": "",
      "courses": Array [
        "1",
      ],
      "createdBy": Object {},
      "description": "Unit test items",
      "grades": Array [],
      "renderingType": "assessment",
      "scoring": Object {
        "testItems": Array [
          Object {
            "points": 0,
          },
          Object {
            "points": 0,
          },
        ],
        "total": 0,
      },
      "standardsTag": Object {
        "curriculum": "",
        "standards": Array [],
      },
      "status": "draft",
      "subjects": Array [
        "Math",
      ],
      "tags": Array [],
      "testItems": Array [
        "5bf28847dc39e409216aea02",
        "5bf28d4edc39e409216aea06",
      ],
      "thumbnail": "https://fakeimg.pl/500x135/",
      "title": "New Test2",
    },
  },
  "type": "[tests] create test request",
}
`;

exports[`tests receive test by id should return an action 1`] = `
Object {
  "payload": Object {
    "id": "5bf2a3e0da9400105cac6cf6",
  },
  "type": "[tests] receive test by id request",
}
`;

exports[`tests receive test by id should return an action 2`] = `
Object {
  "payload": Object {
    "id": "5bf2a3e0da9400105cac6cf7",
  },
  "type": "[tests] receive test by id request",
}
`;

exports[`tests receive test by id should return an action 3`] = `
Object {
  "payload": Object {
    "id": "5bf2a3e0da9400105cac6cf8",
  },
  "type": "[tests] receive test by id request",
}
`;

exports[`tests receive tests should return an action 1`] = `
Object {
  "payload": 1,
  "type": "[tests] receive list request",
}
`;

exports[`tests receive tests should return an action 2`] = `
Object {
  "payload": 2,
  "type": "[tests] receive list request",
}
`;

exports[`tests receive tests should return an action 3`] = `
Object {
  "payload": 2,
  "type": "[tests] receive list request",
}
`;

exports[`tests set default test data should return an action 1`] = `
Object {
  "type": "[tests] set default test data",
}
`;

exports[`tests set test data should return an action 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "createBy": Object {
        "email": "andrey@snapwiz.com",
        "firstName": "andrey",
        "lastName": "vasilev",
      },
      "sharing": Array [
        Object {
          "email": "andrey@snapwiz.com",
          "permission": "rxw",
        },
      ],
    },
  },
  "type": "[tests] set test data",
}
`;

exports[`tests update test should return an action 1`] = `
Object {
  "payload": Object {
    "data": undefined,
    "id": Object {
      "analytics": Object {
        "likes": "0",
        "usage": "0",
      },
      "assignments": Array [],
      "collections": "",
      "courses": Array [],
      "createdBy": Object {},
      "description": "",
      "grades": Array [],
      "renderingType": "assessment",
      "scoring": Object {
        "testItems": Array [
          Object {
            "points": 0,
          },
          Object {
            "points": 0,
          },
        ],
        "total": 0,
      },
      "standardsTag": Object {
        "curriculum": "",
        "standards": Array [],
      },
      "status": "draft",
      "subjects": Array [],
      "tags": Array [],
      "testItems": Array [
        "5bf28847dc39e409216aea00",
        "5bf28d4edc39e409216aea04",
      ],
      "thumbnail": "https://fakeimg.pl/500x135/",
      "title": "New Test",
    },
  },
  "type": "[tests] update test request",
}
`;

exports[`tests update test should return an action 2`] = `
Object {
  "payload": Object {
    "data": undefined,
    "id": Object {
      "analytics": Object {
        "likes": "20",
        "usage": "10",
      },
      "assignments": Array [],
      "collections": "",
      "courses": Array [
        "1",
      ],
      "createdBy": Object {},
      "description": "Unit test items",
      "grades": Array [],
      "renderingType": "assessment",
      "scoring": Object {
        "testItems": Array [
          Object {
            "points": 0,
          },
          Object {
            "points": 0,
          },
        ],
        "total": 0,
      },
      "standardsTag": Object {
        "curriculum": "",
        "standards": Array [],
      },
      "status": "draft",
      "subjects": Array [
        "Math",
      ],
      "tags": Array [],
      "testItems": Array [
        "5bf28847dc39e409216aea02",
        "5bf28d4edc39e409216aea06",
      ],
      "thumbnail": "https://fakeimg.pl/500x135/",
      "title": "New Test2",
    },
  },
  "type": "[tests] update test request",
}
`;
